bin/compile

```diff
@@ -108,10 +108,33 @@ expandVer() {
 # Report deprecated versions to user
 # Use after expandVer
 reportVer() {
-    if <"${DataJSON}" jq -e '.Go.Supported | indices(["'${1}'"]) | length | if . > 0 then true else false end' &> /dev/null; then
+    local ver="${1}"
+
+    if [ "${TOOL}" = "gomodules" ]; then
+        warn ""
+        warn "Go modules are an experimental feature of go1.11"
+        warn "Any issues building code that uses Go modules should be"
+        warn "reported via: https://github.com/heroku/heroku-buildpack-go/issues"
+        warn ""
+        warn "Additional documentation for using Go modules with this buildpack"
+        warn "can be found here: https://github.com/heroku/heroku-buildpack-go#go-module-specifics"
+        warn ""
+        if ! <"${DataJSON}" jq  -e '.Go.SupportsModuleExperiment | any(. == "'${ver}'")' &> /dev/null; then
+            err "You are using ${ver}, which does not support the Go modules experiment"
```

Comment:
`which does not support` => `which does not enable`?

Reply:
IMO `support` is a more accurate word than `enable` here. If we've gotten here then we've already detected a `go.mod` file and selected go modules as the way to do the install, but either because of the release state of go1.11 or their specified go version they/the buildpack has chosen a version of go that does not support go modules. The simplest thing is for us to error out. We *could* attempt to re-run tool detection w/o gomodule support and start again but that would be a really large change and I'd prefer not to have to do that unless we can prove it's necessary.

Reply:
:+1: